### ISSUE TITLE
Firefox: Summarize the current page using an extension shortcut doesn't work

### ISSUE THREAD

**Issue by ybc37** [kagisearch/browser_extensions#31]
Hey,

I've set an extension shortcut for "Summarize the currently active page". When using this shortcut, the browser window for the summary shows up, but it doesn't load the summary. In contrary, it works if I open the extension popup and click on the button "Summary".

Error:

```
Uncaught (in promise) ReferenceError: url is not defined
    requestPageSummary moz-extension://<uuid>/src/summarize_result.js:100
    setup moz-extension://<uuid>/src/summarize_result.js:128
    EventListener.handleEvent* moz-extension://<uuid>/src/summarize_result.js:131
summarize_result.js:100:7
```

Skimming the code (function `requestPageSummary()`), I noticed that in case I use the shortcut, `window.location.search` is empty (without using the shortcut it is something like `?url=https%3A%2F%2Fblog.kagi.com%2Funlimited-searches-for-10&summary_type=summary&target_language=&token=<token>&api_token=&api_engine=cecil`). So it tries to get the required parameters from the active tab (`getActiveTab()`), but `url` is never defined/set.

```js
// [...]
const searchParams = new URLSearchParams(window.location.search); // `searchParams.size` is 0

    // If there's no URL, get the currently active tab and default params
    if (!searchParams.get('url')) {
      searchParams.set('summary_type', 'summary');
      searchParams.set('target_language', '');

      const tab = await getActiveTab();

      if (!tab) {
        console.error('No tab/url found.');
        return;
      }

      searchParams.set('url', url); // `url` is undefined
      // [...]
```

---

Environment:

* Firefox Developer Edition, 118.0b9
* Kagi Search for Firefox, 0.3.7

**Comment by ybc37** [comment]
Actually, it seems like #30 also fixes this? I didn't test it yet, though.

**Comment by BrunoBernardino** [comment]
Nice catch @ybc37 yes, #30 fixed it and it'll be available with 0.3.8.
